doc/dev/osd_internals/log_based_pg.rst

```diff
@@ -81,40 +81,28 @@ cannot.  It would be really annoying if we needed to have two entire
 implementations of PrimaryLogPG, one for each of the two, if there are
 really only a few fundamental differences:
 
-  1. How reads work -- async only, requires remote reads for ec
-  2. How writes work -- either restricted to append, or must
-     write aside and do a tpc
-  3. Whether we choose the oldest or newest possible head entry
-     during peering
-  4. A bit of extra information in the log entry to enable rollback
+#. How reads work -- async only, requires remote reads for ec
+#. How writes work -- either restricted to append, or must write aside and do a tpc
+#. Whether we choose the oldest or newest possible head entry during peering
+#. A bit of extra information in the log entry to enable rollback
 
 and so many similarities
 
-  1. All of the stats and metadata for objects
-  2. The high level locking rules for mixing client IO with recovery
-     and scrub
-  3. The high level locking rules for mixing reads and writes without
-     exposing uncommitted state (which might be rolled back or
-     forgotten later)
-  4. The process, metadata, and protocol needed to determine the set
-     of osds which partcipated in the most recent interval in which we
-     accepted writes
-  5. etc.
-
-Instead, we choose a few abstractions (and a few kludges) to paper
-over the difference:
-
-  1. PGBackend
-  2. PGTransaction
-  2. PG::choose_acting chooses between calc_replicated_acting and
-     calc_ec_acting
-  3. Various bits of the write pipeline disallow some operations based
-     on pool type -- like omap operations, class operation reads, and
-		 writes which are not aligned appends (officially, so far)
-     for ec
-  4. Misc other kludges here and there
-
-PGBackend and PGTransaction enables abstraction of differences 1, 2,
+#. All of the stats and metadata for objects
+#. The high level locking rules for mixing client IO with recovery and scrub
+#. The high level locking rules for mixing reads and writes without exposing uncommitted state (which might be rolled back or forgotten later)
```

Comment:
yes  it works.

but Bullet Lists can't break lines, I use the line-blocks only for "-". Take a look please